src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ProjectState.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Testing
+{
+    public abstract class ProjectState
+    {
+        private readonly string _name;
+        private readonly string _defaultPrefix;
+        private readonly string _defaultExtension;
+
+        protected ProjectState(string name, string defaultPrefix, string defaultExtension)
+        {
+            _name = name;
+            _defaultPrefix = defaultPrefix;
+            _defaultExtension = defaultExtension;
+
+            Sources = new SourceFileList(defaultPrefix, defaultExtension);
+        }
+
+        public string Name { get; set; }
```

Comment:
Its not obvious to me why you can set this value after creating the object

Reply:
I'll make it read only